### DISCUSSION TITLE
Splitting into separate packages

### DISCUSSION THREAD

**Discussion by nibra** [joomla-framework/utilities#37] [General]
The tools in this package are not depending on each other. They could as well be stand-alone packages.
`joomla-framework/utilities` would still exist, but then become a meta package.

### Pro

* Each tool gets its own version number, so adding new tools does not require incrementing the minor version.
* Separate packages encourage other projects to use these tools, as they don't have to install ArrayHelper if they just need IpHelper.

### Con

* The Framework provides more packages, probably requiring more maintenance efford.

Please add your 2 cents, pros and cons from your perspective, so we can come to an educated decision.

**Comment by HLeithner** [comment]
I personal see not a good reason to have a package for each file.

**Reply by nibra** [reply]
Do you see a good reason against?

**Reply by HLeithner** [reply]
More work then needed ;-)

**Comment by rdeutz** [comment]
Makes sense to me, both packages have nothing in common besides that they utlities
